File: benchmarks/verification-stack-v2/typescript/src/service.ts
export async function fetchCount(): Promise<number> {
  return 1;
}

export function calculateTotal(amount: number, rate: number): number {
  return amount * (1 + rate);
}

export function renderTotal(total: number): string {
  return total.toFixed(2);
}

export function increment(count: number): number {
  return count + 1;
}

export async function nextCount(): Promise<number> {
  const count = await fetchCount();
  return count + 1;
}

export const exampleTotal = calculateTotal(100, 0.2);

File: benchmarks/verification-stack-v2/python/src/verification_fixture/service.py
from __future__ import annotations


async def fetch_count() -> int:
    return 1


def calculate_total(amount: float, rate: float) -> float:
    return amount * (1.0 + rate)


def render_total(total: float) -> str:
    return f"{total:.2f}"


def increment(count: int) -> int:
    return count + 1


async def next_count() -> int:
    count = await fetch_count()
    return count + 1


EXAMPLE_TOTAL = calculate_total(100.0, 0.2)
